### PR TITLE
Stop referencing removed attribute in Rails 5.1

### PR DESCRIPTION
**Pull request by martinvium**
This PR completely removes support for connection_id which was removed
from Rails 5.1.

https://github.com/rails/rails/commit/4565de0da14a4b3d75e8b1a652fcb6dcd947c451

This will probably break with older versions of Rails, so 1.1.1 will
probably be the last to support versions older than 5.1.

Also publishes version 1.1.2

### CODE REVIEW

File: lib/query_cache/version.rb [hunk 1]
```diff
@@ -1,3 +1,3 @@
 module QueryCache
-  VERSION = "1.1.1"
+  VERSION = "1.1.2"
```
Comment (bestie) [inline lib/query_cache/version.rb]: I understand this version increment to mean that this is backwards compatible with all supported versions of Rails and upgrading shouldn't break for anyone.

The version in the gem spec says we support Rails 3.1 up 😄

**Comment by lukelex** [comment]
@martinvium @bestie it's ok to end support for `< 5.x`.

As far as I know we don't use it anywhere else but MetadataStore and we just updated to rails 5 and temporarily removed it until we make it work there.

File: lib/query_cache/version.rb (hunk 1 above)
Comment (martinvium) [reply lib/query_cache/version.rb]: Hah it was the minimum change required to Make-It-Work. I will change it to 1.2.0 and update the gemspec 👍

Comment (martinvium) [reply lib/query_cache/version.rb]: Or should I just make it 2.0 and get it over with?

**Comment by martinvium** [comment]
I also considered adding <= 5.0 to activerecord as a dependency in 1.1.1 to to make it explicit. Right now you just get the error when you hit production where caching is enabled.

File: lib/query_cache/version.rb (hunk 1 above)
Comment (lukelex) [reply lib/query_cache/version.rb]: If you don't wanna commit to older versions of Rails just set it to `2.0`

File: lib/query_cache.rb [hunk 2]
```diff
@@ -44,8 +43,7 @@ def base
       self.class.base
     end
 
-    def restore_query_cache_settings(connection_id, enabled)
-      base.connection_id = connection_id
```
Comment (lukelex) [inline lib/query_cache.rb]: Has the connection tracking been replaced with something else in rails 5, or just completely removed?

Comment (martinvium) [reply lib/query_cache.rb]: I can only really speculate about the original intent of connection_id, but reading this code, and the commit in the Rails framework leads me to believe there was an `id` for connections, but it wasn't really used for anything. When dealing with connections it _seems_ to always happen at an "instance" level, without any explicit identifier.

https://github.com/rails/rails/commit/4565de0da14a4b3d75e8b1a652fcb6dcd947c451
> This method appears to have been partially used in connection pool
> caching, but it was introduced without much reasoning or any tests. One
> edge case test was added later on, but it was focused on implementation
> details. This method is no longer used outside of tests, and as such is
> removed.

**Comment by martinvium** [comment]
It's probably worth noting, that this file says it originally came from QueryCache in ActiveRecord.

This file has changed quite a bit in Rails 5.x, though it seems to serve the same purpose:

https://github.com/rails/rails/blob/5-1-stable/activerecord/lib/active_record/query_cache.rb

I considered rewriting this module for forward compatibility, but opted to go for what works for now.
